### PR TITLE
Add support for config files in downloader.

### PR DESCRIPTION
**Pull request by s-l-teichmann**
First step towards storing options in configuration files.

Part of  #396

### CODE REVIEW

File: cmd/csaf_downloader/config.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	"github.com/mitchellh/go-homedir"
+)
+
+const defaultWorker = 2
+
+type config struct {
+	Directory            *string  `short:"d" long:"directory" description:"DIRectory to store the downloaded files in" value-name:"DIR"`
+	Insecure             bool     `long:"insecure" description:"Do not check TLS certificates from provider"`
+	IgnoreSignatureCheck bool     `long:"ignoresigcheck" description:"Ignore signature check results, just warn on mismatch"`
+	Version              bool     `long:"version" description:"Display version of the binary"`
```
Comment (ThomasJunk) [inline cmd/csaf_downloader/config.go:25]: Is this necessary for the ini-file. The output of the version number seems to be suppressed when provided in the config. Tested with
```
Insecure = true
Version = true
Verbose = true
``` 
and a call `./csaf_downloader localhost`

File: cmd/csaf_downloader/config.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	Rate                 *float64 `long:"rate" short:"r" description:"The average upper limit of https operations per second (defaults to unlimited)"`
+	Worker               int      `long:"worker" short:"w" description:"NUMber of concurrent downloads" value-name:"NUM"`
+
+	ExtraHeader http.Header `long:"header" short:"H" description:"One or more extra HTTP header fields"`
+
+	RemoteValidator        string   `long:"validator" description:"URL to validate documents remotely" value-name:"URL"`
+	RemoteValidatorCache   string   `long:"validatorcache" description:"FILE to cache remote validations" value-name:"FILE"`
+	RemoteValidatorPresets []string `long:"validatorpreset" description:"One or more presets to validate remotely" default:"mandatory"`
+
+	Config *string `short:"c" long:"config" description:"Path to config ini file" value-name:"INI-FILE" no-ini:"true"`
```
Comment (ThomasJunk) [inline cmd/csaf_downloader/config.go:36]: Is this part of the expected values within an `.ini`?

Comment (s-l-teichmann) [reply cmd/csaf_downloader/config.go:36]: @ThomasJunk Have You noticed the `no-ini:"true"`` part of the struct tag?

Comment (ThomasJunk) [reply cmd/csaf_downloader/config.go:36]: I was confused what is expected to be in an `.ini`and what is only for the commandline. It was not clear to me.

File: cmd/csaf_downloader/config.go (hunk 1 above)
Comment (s-l-teichmann) [reply cmd/csaf_downloader/config.go]: Commit e9cd31aad1e61deb4e9439cd9e3373e8f54d08b9 adds a `no-ini:"true"` for the version flag, too.

Comment (ThomasJunk) [reply cmd/csaf_downloader/config.go]: Didn't notice the flag.

**Review by ThomasJunk — DISMISSED** [review]
LGTM and works.

**Review by JanHoefelmeyer — APPROVED** [review]
LGTM

**Review by ThomasJunk — APPROVED** [review]
LGTM

**Pull request merged**